### ISSUE TITLE
heredoc not supported after brace group

### ISSUE THREAD

**Issue by TimothyGu** [koalaman/shellcheck#458]
```bash
{ cat <<EOF; }
abc
EOF
```

is a valid shell script, but fails to parse with shellcheck.net:

```
   1  { cat <<EOF; }
      ^––SC1073 Couldn't parse this brace group.
   2  abc
   3  EOF
      ^––SC1056 Expected a '}'. If you have one, try a ; or \n in front of it.
      ^––SC1072 Missing '}'. Fix any mentioned problems and try again.
```

**Comment by TimothyGu** [comment]
```sh
{ cat <<EOF;
abc
EOF
}
```

works fine.

**Comment by kurahaupo** [comment]
"Something else works fine" is not the point; it is valid for the start-mark of a here-doc to be followed by other words on the same line before the contents of the here-doc start at the beginning of the next line. In particular, multiple heredocs can be specified on the same line:

    if ((somecondition)) ; then cat <&3; cat <<E1 ; else cat <<E2 ; cat <&3 ; fi 3<<E3
    stuff for when somecondition is true
    E1
    stuff for when somecondition is false
    E2
    stuff for both cases
    E3

So shellcheck should support it.

(There is an exception where ```$( ... <<EOF ... )``` does not have to scan for the closing ```EOF``` beyond the closing bracket because it may occur inside a quoted string.)

**Comment by TimothyGu** [comment]
> "Something else works fine" is not the point

… which is why I kept this issue open.

**Comment by koalaman** [comment]
Nice. This is another side of the single, oldest bug in shellcheck, and is caused by the parser fundamentally not handling elements occupying disjoint spans.

**Comment by koalaman** [comment]
Fixed in 2f0ae44, wooo
